### ISSUE TITLE
System.NotSupportedException for .Net Core 3.1.3

### ISSUE THREAD

**Issue by bezzad** [obfuscar/obfuscar#268]
When I try to obfuscate a `MyApp.dll` of the `.Net Core 3.1`, I get the following message:

```
Unhandled Exception: System.NotSupportedException: This .resources file should not be read with this reader. The resource reader type is "System.Resources.Extensions.DeserializingResourceReader, System.Resources.Extensions, Version=4.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51".
   at System.Resources.ResourceReader._ReadResources()
   at System.Resources.ResourceReader.ReadResources()
   at Obfuscar.Obfuscator.GetXamlDocuments(AssemblyDefinition library)
   at Obfuscar.Obfuscator.RenameTypes()
   at Obfuscar.Obfuscator.RunRules()
   at Obfuscar.Program.Main(String[] args)
```

My obfuscar config is like this:
```
<?xml version="1.0" encoding="utf-8" ?> 
<Obfuscator>
  <Var name="InPath" value="..\..\output\release\netcoreapp3.1" />
  <Var name="OutPath" value="$(InPath)\Obfuscated" />
  <Var name="KeepPublicApi" value="false" />
  <Var name="HidePrivateApi" value="true" /> 
  
  <Module file="$(InPath)\MyApp.dll" />
</Obfuscator>
```

Do you support this version of `.Net Core` or not?

**Comment by schmidtjohannes** [comment]
I created a docker image and when obfuscar.console it returns:
```
It was not possible to find any compatible framework version
The framework 'Microsoft.NETCore.App', version '2.1.0' was not found.
  - The following frameworks were found:
      3.1.2 at [/usr/share/dotnet/shared/Microsoft.NETCore.App]

You can resolve the problem by installing the specified framework and/or SDK.

The specified framework can be found at:
  - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=2.1.0&arch=x64&rid=debian.10-x64
```

here the dockerfile:
```
FROM mcr.microsoft.com/dotnet/core/sdk:3.1

ENV DOTNET_OBFUSCAR_VERSION=2.2.26
RUN dotnet tool install --global Obfuscar.GlobalTool --version ${DOTNET_OBFUSCAR_VERSION}

ENV PATH="/root/.dotnet/tools:${PATH}"
```

**Comment by lextm** [comment]
@schmidtjohannes you hit #255. Install 2.1 runtime in your Docker image, and the issue should go away.

**Comment by lextm** [comment]
@bezzad do you have a small sample project to reproduce this issue?

**Comment by schmidtjohannes** [comment]
> @schmidtjohannes you hit #255. Install 2.1 runtime in your Docker image, and the issue should go away.

Hi @lextm , did so and the process itself worked. But when installing the dependency (it is a nuget in my case) it doesnt recognize it.

`error CS0234: The type or namespace name 'Bla' does not exist in the namespace 'MyNamespace' (are you missing an assembly reference?)`

**Comment by polarlightning** [comment]
I have maybe related case. I am trying to obfuscate net462 executable in a Docker "builder-container" (Linux container), which is based on Microsoft's base image "mcr.microsoft.com/dotnet/core/sdk:3.1", but Obfuscar is not even starting. When I try to run obfuscar.console (2.2.15) there asking just its version with -V, the below are emitted, indicating, AFAIK, that Obfuscar could run only in .NET Core 2.1.0. Obfuscar was installed in the image with "dotnet tool install --framework netcoreapp3.1 --global Obfuscar.GlobalTool --version 2.2.15", but it seems --framework -parameter is ineffective here (did not make difference to output9.

`# ls -l /root/.dotnet/tools
-rwxr-xr-x 1 root root 86584 May 12 16:34 obfuscar.console

# /root/.dotnet/tools/obfuscar.console -V
_It was not possible to find any compatible framework version
The framework 'Microsoft.NETCore.App', version '2.1.0' was not found.
  - The following frameworks were found:
      3.1.3 at [/usr/share/dotnet/shared/Microsoft.NETCore.App]

You can resolve the problem by installing the specified framework and/or SDK.

The specified framework can be found at:
  - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=2.1.0&arch=x64&rid=debian.10-x64
# exit_`

**Comment by lextm** [comment]
@polarlightning Please read #255 . That's a different issue.

**Comment by molesmoke** [comment]
I'm also seeing this behavior. For me, it manifests in a winforms projects when a form's localizable property is set to true.
Please find a small test project attached that reproduces the issue.

[WindowsFormsApp2.zip](https://github.com/obfuscar/obfuscar/files/5514743/WindowsFormsApp2.zip)

**Comment by techaimail** [comment]
We have the same problem in a WPF .NET 5 app with VStudio 2019 Enterprise, please is there any solution?
We are using Obfuscar 2.2.29

```
Loading assemblies...Extra framework folders: Done.
Hiding strings...

Unhandled Exception: System.NotSupportedException: This .resources file should not be read with this reader. The resource reader type is "System.Resources.Extensions.DeserializingResourceReader, System.Resources.Extensions, Version=4.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51".
   at System.Resources.ResourceReader._ReadResources()
   at System.Resources.ResourceReader.ReadResources()
   at Obfuscar.Obfuscator.GetXamlDocuments(AssemblyDefinition library)
   at Obfuscar.Obfuscator.RenameTypes()
   at Obfuscar.Obfuscator.RunRules()
   at Obfuscar.Program.Main(String[] args)
Renaming:  fields...Parameters...Properties...Events...Methods...Types...
```

**Comment by lextm** [comment]
Resolved by #324
